R/plotEGRET.R

```diff
@@ -0,0 +1,30 @@
+#' Create an EGRET-style plot
+#'  
+#' Use a loadflex object, but create an EGRET plot.
+#' 
+#' @param intdat data.frame of interpolation data
+#' @param estdat data.frame of estimation data
+#' @param preds data.frame of load predictions
+#' @param meta loadflex metadata object; it must include constituent,
+#' flow, conc.units, custom (station abbreviation: staAbbr)
+#' @param plotName the name of the plot the user wants to create. Current
+#' options: ConcTime, ConcQ, FluxTimeDaily, and FluxQ.
+#' @param ... additional arguments to pass to the plot
+#'
+#' @importFrom EGRET plotConcTime
+#' @importFrom EGRET plotConcQ
+#' @importFrom EGRET plotFluxTimeDaily
+#' @importFrom EGRET plotFluxQ
+#'
+#' @export
+plotEGRET <- function(intdat, estdat, preds, meta, plotName, ...){
+  
+  egretobj <- convertToEGRET(intdat, estdat, preds, meta)
+  
+  switch(plotName,
```

Comment:
3 defensive coding moves we can make here to help the user pass in a valid plot name:
1. give `plotName` (soon to be `plot.name`) values in the function declaration that match up with the valid plotting options. this looks like:
```r
plotEGRET <- function(intdat, estdat, preds, meta, plot.name=c('ConcTime', 'ConcQ', 'FluxTimeDaily', 'FluxQ', ...) {
```
2. use match.arg to either confirm that they passed in a valid option, or fall back to the first one as the default
```r
plot.name <- match.arg(plot.name)
```
3. include a default argument to switch that breaks if they still somehow didn't select a valid option. (in combination with the above changes, this one is more a defense against us making inconsistent updates in the future rather than a defense against a bad user input).
```r
switch(plot.name,
  ConcTime = plotConcTime(egretobj, ...),
  ConcQ = plotConcQ(egretobj, ...),
  FluxTimeDaily = plotFluxTimeDaily(egretobj, ...),
  FluxQ = plotFluxQ(egretobj, ...),
  default = stop(paste('unrecognized plot.name:', plot.name)))
```